gateway/src/main.rs

```diff
@@ -129,6 +132,30 @@ async fn main() {
         Arc::new(Config::default())
     };
 
+    // Note - we only enable OTLP after config file parsing/loading is complete,
+    // so that the config file can control whether OTLP is enabled or not.
+    // This means that any tracing spans created before this point will not be exported to OTLP.
+    // For now, this is fine, as we only ever export spans for inference/batch/feedback requests,
+    // which cannot have occurred up until this point.
+    // If we ever want to emit earlier OTLP spans, we'll need to come up with a different way
+    // of doing OTLP initialization (e.g. buffer spans, and submit them once we know if OTLP should be enabled).
+    // See `build_opentelemetry_layer` for the details of exactly what spans we export.
+    if config.gateway.export.otlp.traces.enabled {
+        if std::env::var("OTEL_EXPORTER_OTLP_TRACES_ENDPOINT").is_err() {
+            // This makes it easier to run the gateway in local development and CI
+            if cfg!(feature = "e2e_tests") {
+                tracing::warn!("Running without explicit `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` env var in e2e tests mode")
+            } else {
+                tracing::error!("[gateway.export.otlp.traces] has `enabled = true`, but env var `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` is not set. Please set it to the OTLP endpoint (e.g. `http://localhost:4317`)");
+                std::process::exit(1);
+            }
+        }
+        otel_handle
+            .enable_otel()
+            .expect_pretty("Failed to enable OpenTelemetry");
+        tracing::info!("Enabled OpenTelemetry OTLP export");
```

Comment:
`Enabled OpenTelemetry OTLP export to {endpoint}`

Reply:
This is tricky, as the OTEL sdk can actually look at different environment variables (e.g. `OTEL_EXPORTER_OTLP_ENDPOINT`). Unfortunately, `opentelemetry-otlp` doesn't seem to expose the actual URL that it's using.

Reply:
Ah no worries then.